code/__HELPERS/_logging.dm

```diff
@@ -109,10 +117,12 @@ GLOBAL_PROTECT(log_end)
 		rustg_log_write(GLOB.world_game_log, "ADMINWARN: [html_decode(text)][GLOB.log_end]")
 
 /proc/log_pda(text, mob/speaker)
+	check_for_troll("PDA", text, speaker)
 	if(config.log_pda)
 		rustg_log_write(GLOB.world_game_log, "PDA: [speaker.simple_info_line()]: [html_decode(text)][GLOB.log_end]")
 
 /proc/log_chat(text, mob/speaker)
+	check_for_troll("Chat", text, speaker)
```

Comment:
Each use of the `check_for_troll` proc returns either TRUE or FALSE. However, there is not if statement surrounding the proc calls to return out of the "messaging" procs. So, if someone uses a "gamer" word, they will get banned, but that message will also go through, right?

Instead, shouldn't it be:

Reply:
```
/proc/log_chat(text, mob/speaker)
	if(config.log_pda && !check_for_troll("Chat", text, speaker)
		rustg_log_write(GLOB.world_game_log, "CHAT: [speaker.simple_info_line()] [html_decode(text)][GLOB.log_end]")
```
?

Reply:
Ah, no, that goes to the logs. Got it :)